test/SparqlParser-test.js

```diff
@@ -34,56 +40,62 @@ describe('A SPARQL parser', function () {
     try { parser.parse(query); }
     catch (e) { error = e; }
 
-    expect(error).to.exist;
-    expect(error).to.be.an.instanceof(Error);
-    expect(error.message).to.include('Parse error on line 1');
+    expect(error).not.toBeUndefined();
+    expect(error).toBeInstanceOf(Error);
+    expect(error.message).toContain('Parse error on line 1');
   });
 
   it('should preserve BGP and filter pattern order', function () {
-    var parser = new SparqlParser(null, null, factory);
+    var parser = new SparqlParser(null, null, null);
     var query = 'SELECT * { ?s ?p "1" . FILTER(true) . ?s ?p "2"  }';
     var groups = parser.parse(query).where;
-    expect(groups[0].type).to.equal("bgp");
-    expect(groups[1].type).to.equal("filter");
-    expect(groups[2].type).to.equal("bgp");
+    expect(groups[0].type).toBe("bgp");
+    expect(groups[1].type).toBe("filter");
+    expect(groups[2].type).toBe("bgp");
   });
 
   describe('with pre-defined prefixes', function () {
     var prefixes = { a: 'ex:abc#', b: 'ex:def#' };
-    var parser = new SparqlParser(prefixes, null, factory);
+    var parser = new SparqlParser(prefixes, null, null);
 
     it('should use those prefixes', function () {
       var query = 'SELECT * { a:a b:b "" }';
-      expect(parser.parse(query).where[0].triples[0])
-        .to.deep.equal({subject: 'ex:abc#a', predicate: 'ex:def#b', object: '""'});
+      var result_json = '{"subject":{"termType":"NamedNode","value":"ex:abc#a"},"predicate":{"termType":"NamedNode","value":"ex:def#b"},"object":{"termType":"Literal","value":"","language":"","datatype":{"termType":"NamedNode","value":"http://www.w3.org/2001/XMLSchema#string"}}}';
```

Comment:
I suggest skipping the JSON parsing, and just entering this as JS objects directory. (Also later on)